[receiver/gitproviderreceiver] add Pull Request metrics (#29642)
open-telemetry/opentelemetry-collector-contrib: receiver/gitproviderreceiver/documentation.md

| ---- | ----------- | ---------- |
| 1 | Gauge | Int |

### git.repository.pull_request.approved.time

The amount of time it took a pull request to go from open to approved

| Unit | Metric Type | Value Type |
| ---- | ----------- | ---------- |

> | 1 | Gauge | Int |

Reviewer: Looks like all the metric unit fixes have been lost somewhere in the rebases/merges? See [this comment](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/29642#pullrequestreview-1794151974) on this PR and the changes below that comment.

```suggestion
| s | Gauge | Int |
```

#### Attributes

| Name | Description | Values |
| ---- | ----------- | ------ |
| repository.name | The full name of a Git repository | Any Str |
| branch.name | The name of a Git branch | Any Str |

### git.repository.pull_request.merged.count

The number of merged pull requests in a repository

| Unit | Metric Type | Value Type |
| ---- | ----------- | ---------- |
| 1 | Gauge | Int |

#### Attributes

| Name | Description | Values |
| ---- | ----------- | ------ |
| repository.name | The full name of a Git repository | Any Str |

### git.repository.pull_request.merged.time

The amount of time it took a pull request to go from open to merged

| Unit | Metric Type | Value Type |
| ---- | ----------- | ---------- |
| 1 | Gauge | Int |

#### Attributes

| Name | Description | Values |
| ---- | ----------- | ------ |

> | repository.name | The full name of a Git repository | Any Str |

Reviewer 2: ```suggestion
| repository.name | The name of a Git repository | Any Str |
```

| branch.name | The name of a Git branch | Any Str |

### git.repository.pull_request.open.count

The number of open pull requests in a repository

| Unit | Metric Type | Value Type |
| ---- | ----------- | ---------- |
| 1 | Gauge | Int |

#### Attributes

| Name | Description | Values |
| ---- | ----------- | ------ |

> | repository.name | The full name of a Git repository | Any Str |

Reviewer 2: I think we agreed to not say `the full name` and instead just say `The name of a Git repository`.

### git.repository.pull_request.open.time

The amount of time a pull request has been open

| Unit | Metric Type | Value Type |
| ---- | ----------- | ---------- |
| 1 | Gauge | Int |

#### Attributes

| Name | Description | Values |
| ---- | ----------- | ------ |

> | repository.name | The full name of a Git repository | Any Str |

Reviewer 2: I think we agreed to not say `the full name` and instead just say `The name of a Git repository`.

| ---- | ----------- | ---------- |
| 1 | Gauge | Int |

### git.repository.pull_request.approval.time

> The amount of time for a PR to be approved

Reviewer 2: ```suggestion
The amount of time it took for a pull request to be approved.
```
